src/Function.h

```diff
@@ -204,6 +215,25 @@ class Function {
     /** Check if a function has been frozen. If so, it is an error to
      * add new definitions. */
     EXPORT bool frozen() const;
+
+    /** Mark calls of this function by 'f' to be replaced with its wrapper
+     * during the lowering stage. If the string 'f' is empty, it means replace
+     * all calls to this function by all other functions (excluding itself) in
+     * the pipeline with the wrapper. See \ref Func::in for more details. */
+    EXPORT void add_wrapper(const Function &wrapper, const std::string &f);
+
+    /* Return map of wrappers defined on this function. */
+    const std::map<std::string, IntrusivePtr<Internal::FunctionContents>> &wrappers() const;
+
+    /** Replace every call to function in 'substitutions' keys (including calls
+     * in each of its update's RDom's predicates) to call to function in
+     * 'substitutions' values. */
+    EXPORT Function &substitute_calls(const std::map<Function, Function> &substitutions);
```

Comment:
To make a doxygen comment encompass multiple methods, wrap the methods in //@{  and //@}